withastro/astro
# Filter out Svelte's unknown class prop console warnings

## Changes

Astro's hydration code passes a `class` prop to Svelte components, inducing Svelte to log a warning about an unknown prop.

Preempting this by exporting a `class` prop from the Svelte component isn't a viable workaround since `class` is a reserved identifier in JS.

This PR implements the console-filtering workaround suggested by @HiDeoo in #5665, borrowing the `useConsoleFilter` approach from the [preact integration](https://github.com/withastro/astro/blob/a1c0cbe604c9f91cdc421b5606aab574999eba01/packages/integrations/preact/src/server.ts#L72-L94).

It would probably be better to generalize the console filtering process so it could be shared across multiple integrations.

Ideally there would be a way to handle this in Svelte, but as was pointed out in the issue thread even they resort to [similar hackery](https://github.com/sveltejs/kit/blob/master/packages/kit/src/runtime/client/client.js#L1974-L1996) in SvelteKit.

Note that you may need to clear your Vite cache to get this change to stick:
```sh
rm -rf ./node_modules/.vite/
```

## Testing

1. Create an Astro + Svelte project: 
    ```sh
    npm create astro@latest -- --template framework-svelte
    ````

2. Run the project and observe the browser console's warning output:
    ```sh
    npm run dev
    ```

    Console output:
    ```
    <Counter> was created with unknown prop 'class'
    ```

3. Apply the patch

4. Clear the vite cache:
    ```sh
    rm -rf ./node_modules/.vite/
    ```

2. Run the project and observe the browser console's (lack of) warning output:
    ```sh
    npm run dev
    ```

    Console output:
    ```
    ```

## Docs

This is a non-breaking change that offers a minor quality of life improvement, not sure documentation beyond what's included in the source comments is necessary.
